app.py

```diff
@@ -155,19 +214,32 @@ def baremetal_order_fulfill():
         if not order_data:
             return jsonify({'error': 'Missing order data'}), 400
 
+        order_id = order_data.get('order_id')
         network_id = order_data.get('network_id')
         requested_nodes = order_data.get('nodes')
-        if not network_id or not requested_nodes:
-            return jsonify({'error': 'Missing network_id or nodes'}), 400
+        if not order_id or not network_id or not requested_nodes:
+            return jsonify({'error': 'Missing order_id, network_id or nodes'}), 400
 
         # Verify if the network exists
         conn = get_openstack_connection(cloud=cloud_name)
         network_res = conn.network.find_network(network_id)
         if not network_res:
             return jsonify({'error': f'Network "{network_id}" not found'}), 404
 
+        # Send message to Kafka topic to kick off fulfillment
+        producer.send(kafka_topic_order_loop, json.dumps(order_data).encode('utf-8'))
+        LOG.info(f"Sent the baremetal order fulfillment request to Kafka: {order_data}")
+
+        zk = KazooClient(hosts='%s:%s' % (zookeeper_host_name, zookeeper_port))
+        zk.start()
+        try:
+            zk.create(f"baremetal/fulfill_order/Report/{order_id}/reportStatus", bytes("Running", "utf-8"), makepath=True)
+        except NodeExistsError as e:
+            zk.set(f"baremetal/fulfill_order/Report/{order_id}/reportStatus", bytes("Running", "utf-8"))
+            pass
+
         # Start fulfillment in background
-        t = Thread(target=run_fulfillment_background, args=(network_id, requested_nodes))
+        t = Thread(target=run_fulfillment_background, args=(network_id, requested_nodes, order_id, zk))
```

Comment:
I suggest instead of creating a thread here, you could use the `producer.send` call above to send a message to the `run_fulfillment_background` function with the `@bus_run.handle(kafka_topic_order_loop)` declared there instead. Just be sure that the message received has the `network_id`, `requested_nodes`, `order_id` values. And you can initialize the zookeeper client inside of `run_fulfillment_background` instead of here. This will remove some unnecessary threading. Would that work?